src/pwm.rs

```diff
@@ -0,0 +1,186 @@
+use crate::println;
+use embedded_hal::Pwm;
+use riot_sys::pwm_t;
+
+/// PWM modes
+///
+/// [`PWMMode::CENTER`] exists in RIOT but is marked as "not supported"
+#[derive(Debug)]
+pub enum PWMMode {
+    LEFT,
+    RIGHT,
+    CENTER,
+}
+
+impl PWMMode {
+    /// Converts the rust enum to a c type
+    pub fn to_c(self) -> riot_sys::pwm_mode_t {
+        match self {
+            Self::LEFT => riot_sys::pwm_mode_t_PWM_LEFT,
+            Self::RIGHT => riot_sys::pwm_mode_t_PWM_RIGHT,
+            Self::CENTER => riot_sys::pwm_mode_t_PWM_CENTER,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct PWMDevice<const CHANNELS: usize> {
+    dev: pwm_t,
+    channel_duty_values: [u16; CHANNELS], //Needed because of embedded_hal implementation
+    frequency: u32,                       //Needed because of embedded_hal implementation
+}
+
+pub type Hz = u32;
```

Comment:
Should work since it seems just to be an wrapper around u32. This would introduce a new dependency to the wrapper crate if I am not mistaken? But if thats ok i would use this type.

Reply:
Adding dependencies is generally OK if they're good ones.

I've had a a closer look at embedded-time now, and it seem like [fugit::rate::Hz](https://docs.rs/fugit/0.3.6/fugit/struct.Rate.html#method.Hz) would be the better candidate even.

(Neither has reviews at https://web.crev.dev/; maybe I'll manage a short one).

Reply:
I've build something using `fugit::HertzU32` in https://github.com/RIOT-OS/rust-riot-wrappers/pull/38/commits/64fcdc30719949b31a94f994a96563df1e16c0a4 . I hope that suffices as a temporary solution until `embedded_hal` changes and all of this needs to be reviewed again.

Reply:
If we think that this type is the right choice we also should add a `pub use fugit` in `lib.rs`

Reply:
Using all of fugit would open up quite a bit of API surface (esp. if fugit people decide to split and keep compatibility re-exports in place), but a `pub use fugit::HertzU32;` in the PWM module would certainly be convenient.